src/test/regress/sql/subselect_gp_indexes.sql

```diff
@@ -0,0 +1,47 @@
+--
+-- Test correlated subquery in subplan with motion chooses correct scan type
+--
+-- Given distributed table
+create table subplan_motion_t
+(user_id bigint NOT NULL) DISTRIBUTED BY (user_id);
+-- with constraint
+alter table only subplan_motion_t add constraint users_choose_tidscan_pkey primary key (user_id);
+-- and some data
+insert into subplan_motion_t
+select gen from generate_series(1, 20000) gen;
+
+-- Check that Index Scan is not used
+select cte."userId" from (select 7 as "userId" ) cte
```

Comment:
May be we can use some lowercase alias instead of "userId" that should be double quoted? And btw, column name "a" seems to be a more general, that "user_id" (the last one seems business logic specific).